activerecord/test/cases/associations/eager_test.rb

```diff
@@ -1511,6 +1511,49 @@ def test_preloading_has_many_through_with_custom_scope
     Author.preload(:readonly_comments).first!
   end
 
+  # Associations::Preloader#preloaders_on works with hash-like objects
+  test "preloading works when object that responds to :to_hash is passed" do
+    post_id = Comment.where(author_id: nil).where.not(post_id: nil).first.post_id
```

Comment:
Does the specific `post_id` matter in these tests? Can we use `Post.preload(...).first` instead?